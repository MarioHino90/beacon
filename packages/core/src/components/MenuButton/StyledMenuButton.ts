import styled, { css } from "styled-components";
import { getColor, getSpacing } from "@braze/beacon-styling";

import { StyledBodyText } from "../BodyText";
import { StyledHeading } from "../Heading";
import { StyledIcon } from "../Icon";

import { StyledMenuButtonProps } from "./types";

export const StyledMenuButton = styled.button<StyledMenuButtonProps>`
  display: flex;
  align-items: center;
  width: 100%;
  padding: ${getSpacing("sm")} ${getSpacing("xl")};
  text-align: left;

  &:hover:not(:disabled),
  &:focus:not(:disabled) {
    background-color: ${getColor("gray", 100)};
  }

  ${StyledIcon} {
    margin-right: ${getSpacing("sm")};
    font-size: 1rem;
  }

  ${(props) =>
    props.disabled &&
    css`
      cursor: not-allowed;

      ${StyledHeading}, ${StyledBodyText}, ${StyledIcon} {
        color: ${getColor("gray", 400)};
      }

      .bcl-braze-icon {
        opacity: 0.3;
      }
    `};
`;
