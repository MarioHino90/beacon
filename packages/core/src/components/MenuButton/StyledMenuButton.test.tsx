import React from "react";
import { shallow } from "enzyme";
import faker from "@faker-js/faker";
import theme from "@braze/beacon-theme";

import { StyledMenuButton } from "./StyledMenuButton";

faker.seed(1603823212124);

describe("StyledMenuButton", () => {
  let props;

  beforeEach(() => {
    props = {
      children: faker.random.words(),
      className: faker.random.word(),
      id: faker.random.word(),
      theme,
    };
  });

  it("should match the snapshot", () => {
    const wrapper = shallow(<StyledMenuButton {...props} />);

    expect(wrapper).toMatchSnapshot();
  });
});
