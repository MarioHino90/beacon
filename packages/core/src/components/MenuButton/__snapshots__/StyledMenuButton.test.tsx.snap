// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StyledMenuButton should match the snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  padding: 0.5rem 1.5rem;
  text-align: left;
}

.c0:hover:not(:disabled),.c0:focus:not(:disabled) {
  background-color: #E9EBED;
}

<button
  className="c0 Direct"
  id="Account"
>
  Borders
</button>
`;
